internal/pkg/skuba/gangway/gangway.go

```diff
@@ -96,21 +90,30 @@ func CreateCert(
 
 	// Generate gangway certificate
 	cert, key, err := util.NewServerCertAndKey(caCert, caKey,
-		CertCommonName, cfg.ClusterConfiguration.APIServer.CertSANs)
+		certCommonName, cfg.ClusterConfiguration.APIServer.CertSANs)
 	if err != nil {
 		return errors.Wrap(err, "could not genenerate gangway server cert")
 	}
 
 	// Create or update certificate to secret
-	if err := util.CreateOrUpdateCertToSecret(client, caCert, cert, key, SecretCertName); err != nil {
+	if err := util.CreateOrUpdateCertToSecret(client, caCert, cert, key, secretCertName); err != nil {
 		return errors.Wrap(err, "unable to create/update cert to secret")
 	}
 
 	return nil
 }
 
-// GetGangwayImage returns gangway image registry
-func GetGangwayImage() string {
-	return images.GetGenericImage(skuba.ImageRepository, imageName,
-		kubernetes.CurrentAddonVersion(kubernetes.Gangway))
+func GangwaySecretExists(client clientset.Interface) (bool, error) {
+	_, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(secretKeyName, metav1.GetOptions{})
+	return kubernetes.DoesResourceExist(err)
+}
+
+func GangwayCertExists(client clientset.Interface) (bool, error) {
```

Comment:
`GangwayCertExists` and `RestartPods` are identical to the ones from the `dex` package. Can't we have a shared package for these two addons or something like that?